src/handlers/CrossMarginHandler02.sol

```diff
@@ -0,0 +1,641 @@
+  /// @param _shouldUnwrap Whether to unwrap WETH into native ETH after withdrawing.
+  /// @return _orderId The ID of the newly created withdraw order.
+  function createWithdrawCollateralOrder(
+    uint8 _subAccountId,
+    address _token,
+    uint256 _amount,
+    uint256 _executionFee,
+    bool _shouldUnwrap
+  ) external payable nonReentrant onlyAcceptedToken(_token) returns (uint256 _orderId) {
+    if (_amount == 0) revert ICrossMarginHandler_BadAmount();
+    if (_executionFee < minExecutionOrderFee) revert ICrossMarginHandler_InsufficientExecutionFee();
+    if (msg.value != _executionFee) revert ICrossMarginHandler_InCorrectValueTransfer();
+    if (_shouldUnwrap && _token != ConfigStorage(CrossMarginService(crossMarginService).configStorage()).weth())
+      revert ICrossMarginHandler_NotWNativeToken();
+
+    // convert native to WNative (including executionFee)
+    _transferInETH();
+
+    // Get the sub-account and order index for the limit order
+    address subAccount = HMXLib.getSubAccount(_msgSender(), _subAccountId);
+    uint256 _orderIndex = withdrawOrdersIndex[subAccount];
+
+    // _orderId = withdrawOrders.length;
+
+    WithdrawOrder memory order = WithdrawOrder({
+      account: payable(msg.sender),
+      orderIndex: _orderIndex,
+      token: _token,
+      amount: _amount,
+      executionFee: _executionFee,
+      shouldUnwrap: _shouldUnwrap,
+      subAccountId: _subAccountId,
+      crossMarginService: CrossMarginService(crossMarginService),
+      createdTimestamp: uint48(block.timestamp),
+      executedTimestamp: 0,
+      status: WithdrawOrderStatus.PENDING
+    });
+
+    _addOrder(order, subAccount, _orderIndex);
+
+    emit LogCreateWithdrawOrder(msg.sender, _subAccountId, _orderId, _token, _amount, _executionFee, _shouldUnwrap);
+    return _orderId;
+  }
+
+  function executeOrders(
+    address[] memory _accounts,
+    uint8[] memory _subAccountIds,
+    uint256[] memory _orderIndexes,
+    address payable _feeReceiver,
+    bytes32[] memory _priceData,
+    bytes32[] memory _publishTimeData,
+    uint256 _minPublishTime,
+    bytes32 _encodedVaas,
+    bool _isRevert
+  ) external nonReentrant onlyOrderExecutor {
+    _executeOrders(
+      _accounts,
+      _subAccountIds,
+      _orderIndexes,
+      _feeReceiver,
+      _priceData,
+      _publishTimeData,
+      _minPublishTime,
+      _encodedVaas,
+      _isRevert
+    );
+  }
+
+  function executeOrders(
+    address[] memory _accounts,
+    uint8[] memory _subAccountIds,
+    uint256[] memory _orderIndexes,
+    address payable _feeReceiver,
+    bytes32[] memory _priceData,
+    bytes32[] memory _publishTimeData,
+    uint256 _minPublishTime,
+    bytes32 _encodedVaas
+  ) external nonReentrant onlyOrderExecutor {
+    _executeOrders(
+      _accounts,
+      _subAccountIds,
+      _orderIndexes,
+      _feeReceiver,
+      _priceData,
+      _publishTimeData,
+      _minPublishTime,
+      _encodedVaas,
+      false
+    );
+  }
```

Comment:
Can remove this. We did this in LimitTradeHandler to stay backward-compatible and not to change code much.
We will keep only with `_isRevert` version.